### ISSUE TITLE
Phyloseq file is not generated

### ISSUE THREAD

**Issue by Jgruetzke** [quadram-institute-bioscience/dadaist2#21]
Hello, 
I recently installed dadaist2 via conda and after adjusting some parameters, the pipeline finished. However, I did get an error for the generation of the phyloseq file. 

ERROR: R script failed with exit code 1
R version 4.0.5 (2021-03-31) 
 * Input:  /cephfs/abteilung4/Projects_NGS/16S/Veganerstudie/dadaist2/Singlets_raw_data/dadaist2_out/ 
 * Loading feature table
 * Taxonomy loaded
 * Tree loaded
 * Metadata loaded
 * PhyloSeq: Feature table done
 * PhyloSeq: Taxonomy table done
 * PhyloSeq: adding tree
Error in validObject(.Object) : invalid class “phyloseq” object: 
 Component sample names do not match.
 Try sample_names()
Calls: phyloseq ... do.call -> new -> initialize -> initialize -> validObject
Execution halted

Is this issue related to the sample naming? 
e.g. 22-SQ00181-0 and VE-17-0109-007-2

Thank you for your help with this problem!
Best,
Josephine

**Comment by telatin** [comment]
Hello and thanks for reportin @Jgruetzke 

You might be right: if you check the log at the beginning you should have got a warning about sample names starting by a digit. It was only a warning because on some cases this is not an issue, but I want to fix this eventually.

Would you be please able to rerun it adding, for example, an "S" in front of the files? Thanks a million

**Comment by Jgruetzke** [comment]
Hi @telatin,

thank you very much for your fast response. I checked with the letter in front of the files, but still got the same error. However, it seems that the dashes in the sample names are a problem here. When I remove those, the phyloseq-file is generated. And also underscores should only be used for the _R1 and _R2, but not in other parts of the sample name, otherwise it is a problem for the metadata sheet generation. 

The automatic sample naming from the fastq files in the metadatasheet using dadaist2-metadata is super convenient, but it would be esp. in this case quite useful to be able to use different sample names without changing the name of the fastq-files. I tested this by replacing the names in the #SampleID column of the metadata file and leaving the fastq names as it was, but it failed at the beginning of the pipeline.

**Comment by telatin** [comment]
Thanks for the feedback!
Yes, metadata is opinionated and the tradeoff between a reasonable metadata generated by the user (to be checked) and one automatically generated can be tricky. THere are flags to modify how the sample names is inferred (separator etc) but the idea would be that SampleName1_R1.fastq.gz would be the cleanest input for dadaist2.

A major rewrite is ongoing, if it sees the light metadata will be very different

**Comment by Jgruetzke** [comment]
Hi @telatin,

thank you very much for your fast response. I checked with the letter in front of the files, but still got the same error. However, it seems that the dashes in the sample names are a problem here. When I remove those, the phyloseq-file is generated. And also underscores should only be used for the _R1 and _R2, but not in other parts of the sample name, otherwise it is a problem for the metadata sheet generation. 

The automatic sample naming from the fastq files in the metadatasheet using dadaist2-metadata is super nice, but it would be esp. in this case quite convenient to be able to use different sample names without changing the name of the fastq-files. I tested this by replacing the names in the #SampleID column and leaving the fastq names as it was, but it failed.
